helm/dex-app/templates/dex/secret.yaml

```diff
@@ -47,32 +47,26 @@ stringData:
       signingKeys: "{{ .Values.oidc.expiry.signingKeys }}"
       idTokens: "{{ .Values.oidc.expiry.idTokens }}"
     staticClients:
-    {{- if (and .Values.oidc.staticClients.gitopsui.clientID  .Values.oidc.staticClients.gitopsui.clientSecret) }}
+    {{- if .Values.oidc.staticClients.gitopsui.clientID }}
     - id: {{ .Values.oidc.staticClients.gitopsui.clientID }}
+      {{- if .Values.oidc.staticClients.gitopsui.clientSecret }}
       secret: {{ .Values.oidc.staticClients.gitopsui.clientSecret }}
+      {{- else }}
+      public: true
+      {{- end }}
       name: gitopsui
       redirectURIs:
       - {{ .Values.oidc.staticClients.gitopsui.redirectURI }}
       - http://localhost:9001/oauth2/callback
       {{- include "trusted-peers" .Values.oidc.staticClients.gitopsui.trustedPeers }}
     {{- end }}
-    {{- if .Values.oidc.staticClients.grafana.clientID }}
-    - id: {{ .Values.oidc.staticClients.grafana.clientID }}
-      public: true
-      redirectURIs:
-      - {{ .Values.oidc.staticClients.grafana.redirectURI }}
-      name: grafana
-      {{- include "trusted-peers" .Values.oidc.staticClients.grafana.trustedPeers }}
-    {{- end }}
-    {{- if .Values.oidc.staticClients.gsCLIAuth.clientID }}
-    - id: {{ .Values.oidc.staticClients.gsCLIAuth.clientID }}
-      public: true
-      name: gscliauth
-      {{- include "trusted-peers" .Values.oidc.staticClients.gsCLIAuth.trustedPeers }}
-    {{- end }}
```

Comment:
Double checking that these removals are intentional -- there are still values for grafana and gsCLI in the values. Maybe double check for references in other files.